core/client/src/main/java/alluxio/client/lineage/AbstractLineageClient.java

```diff
@@ -63,12 +65,19 @@ public long createLineage(List<AlluxioURI> inputFiles, List<AlluxioURI> outputFi
     // TODO(yupeng): relax this to support other type of jobs
     Preconditions.checkState(job instanceof CommandLineJob,
         PreconditionMessage.COMMAND_LINE_LINEAGE_ONLY);
-    LineageMasterClient masterClient = mContext.acquireMasterClient();
+    LineageMasterClient masterClient = null;
     try {
+      masterClient = mContext.acquireMasterClient();
       long lineageId = masterClient.createLineage(stripURIList(inputFiles),
           stripURIList(outputFiles), (CommandLineJob) job);
       LOG.info("Created lineage {}", lineageId);
       return lineageId;
+    } catch (NotFoundException e) {
+      throw new FileDoesNotExistException(e.getMessage());
+    } catch (UnavailableException e) {
+      throw new IOException(e.getMessage());
+    } catch (AlluxioStatusException e) {
+      throw e.toAlluxioException();
```

Comment:
why do we sometimes convert `AlluxioStatusException` to `IOException` (e.g. in `FileOutStream.java`) and other times to `AlluxioException` (e.g. here)?